azurePipeline.yml

```diff
@@ -1,12 +1,3 @@
-# This pipeline should be triggered by every local branches, but should not be used for external PRs.
-# In fact, this pipeline is able to use some of our secrets to push images on dockerhub and deploy on k8s.
-pr: none
-trigger:
-  branches:
-    include:
-    - '*'
```

Comment:
why do you delete this section? The pipeline will no longer be triggered by PRs?

Reply:
It's actually the opposite. The `pr: none` bit prevented the pipeline to be triggered by a PR.  However, we want every PR to trigger the tests. External PRs won't have access to the secrets, thus this is fine. 
By removing all of this, we get the default behavior, which is run for every PR.